.ci/common.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+# Attention, there is no "-x" to avoid problems on Travis
+set -e
+
+# result of command goes to $RUN_JOB
+# 0 = not run (skip)
+# 1 = run (no skip)
+function should_run_job {
+    SKIP_JOB_BY_FILES=$1
+    SKIP_FILES=$2
+    DEBUG=$3
+
+    echo "Initial value of SKIP_JOB_BY_FILES: $SKIP_JOB_BY_FILES"
+    echo "Files to that are allowed to be skipped: $SKIP_FILES"
+
+    if [[ $SKIP_JOB_BY_FILES != 'false' ]]; then
+         ## Identify previous commit to know how much to check
+         ## Check assumes we are only working with 1 commit if we are in master
+         PREVIOUS_COMMIT="HEAD~1"
+
+         if [[ $DEBUG == "true" ]]; then
+              OUT=`git rev-parse HEAD`
+              echo "Head commit: $OUT"
+         fi
+         if [[ $DEBUG == "true" ]]; then
+              OUT=$(git branch master --contains `git rev-parse HEAD`)
+              echo "Master contains head commit: $OUT"
+         fi
+
+         if [[ $(git branch master --contains `git rev-parse HEAD` \
+                    | wc -c ) == 0 ]]; then
```

Comment:
This checks that the current commit we are on is not in master. This was the only way to identify a master build.

For travis, in PR top commit is merge commit. In master CI, we are in a detached master commit.

Reply:
I would place this comment in code.
Bash is not very read-easy code, so comments are helping.

Reply:
Done.